cmd/get_logs.go

```diff
@@ -0,0 +1,352 @@
+	}
+	defer client.Close()
+
+	stdout, err := glc.uploadScript(node, client)
+	if err != nil {
+		return stdout, err
+	}
+	stdout, err = glc.executeScript(node, client)
+	if err != nil {
+		return stdout, err
+	}
+	stdout, err = glc.downloadLogs(node, client)
+	if err != nil {
+		return stdout, err
+	}
+	return "", nil
+}
+
+func (glc *getLogsCmd) uploadScript(node v1.Node, client *ssh.Client) (string, error) {
+	if isWindowsNode(node) || glc.linuxScriptPath == "" {
+		return "", nil
+	}
+
+	scriptContent, err := ioutil.ReadFile(glc.linuxScriptPath)
+	if err != nil {
+		return "", errors.Wrap(err, "reading log collection script content")
+	}
+
+	log.Debugf("Uploading log collection script (%s)\n", glc.linuxScriptPath)
+	session, err := client.NewSession()
+	if err != nil {
+		return "", errors.Wrap(err, "creating SSH session")
+	}
+	defer session.Close()
+
+	session.Stdin = bytes.NewReader(scriptContent)
+	if co, err := session.CombinedOutput("bash -c \"cat /dev/stdin > /tmp/collect-logs.sh\""); err != nil {
+		return fmt.Sprintf("%s -> %s", node.Name, string(co)), errors.Wrap(err, "uploading log collection script")
+	}
+	return "", nil
+}
+
+func (glc *getLogsCmd) executeScript(node v1.Node, client *ssh.Client) (string, error) {
+	log.Debug("Collecting logs\n")
+	session, err := client.NewSession()
+	if err != nil {
+		return "", errors.Wrap(err, "creating SSH session")
+	}
+	defer session.Close()
+
+	var script, cmd string
+	if isLinuxNode(node) {
+		if glc.linuxScriptPath != "" {
+			script = "/tmp/collect-logs.sh"
+			cmd = fmt.Sprintf("bash -c \"sudo chmod +x %s; sudo %s; rm %s\"", script, script, script)
+		} else {
+			script = "/opt/azure/containers/collect-logs.sh"
+			cmd = fmt.Sprintf("bash -c \"sudo %s\"", script)
+		}
+	} else {
+		script = "c:\\k\\debug\\collect-windows-logs.ps1"
+		cmd = fmt.Sprintf("powershell -command \"%s | Where-Object { $_.extension -eq '.zip' } | Copy-Item -Destination $env:temp\\$env:computername.zip\"", script)
+	}
+
+	if co, err := session.CombinedOutput(cmd); err != nil {
+		return fmt.Sprintf("%s -> %s", node.Name, string(co)), errors.Wrap(err, "collecting logs on remote host")
+	}
+	return "", nil
+}
+
+func (glc *getLogsCmd) downloadLogs(node v1.Node, client *ssh.Client) (string, error) {
+	log.Debug("Downloading logs\n")
+	session, err := client.NewSession()
+	if err != nil {
+		return "", errors.Wrap(err, "creating SSH session")
+	}
+	defer session.Close()
+
+	localFileName := fmt.Sprintf("%s.zip", node.Name)
+	localFilePath := path.Join(glc.outputDirectory, localFileName)
+	file, err := os.OpenFile(localFilePath, os.O_WRONLY|os.O_CREATE, 0644)
+	if err != nil {
+		return "", errors.Wrap(err, "opening destination file")
+	}
+	defer file.Close()
+
+	stdout, err := session.StdoutPipe()
+	if err != nil {
+		return "", errors.Wrap(err, "opening ssh session stdout pipe")
+	}
+
+	var cmd string
+	if isLinuxNode(node) {
+		cmd = "bash -c \"cat /tmp/logs.zip > /dev/stdout\""
+	} else {
+		cmd = "type %TEMP%"
```

Comment:
What is this trying to do?

Reply:
Command `type path\to\zip` pipes the zip file content to the SSH session's stdout.